### PR TITLE
Py3 provide http request functionality

### PR DESCRIPTION
**Pull request by tobes**
Seeing as #522 seems to have stalled here is an alternative.

it adds `py3.request()` which returns a simple object that can provide the http status code, text and json.

It allows basic authentication.
request is POST if data is supplied otherwise it will be a GET
query string params can be given in url or as a dict
headers can be supplied.
timeout can be given

I did add proxy support but it ended up getting so complicated (both as code but also testing - https is not supported well) so I removed it all.  For complex things we should just use requests (I'd like to eventually do that via this method so that for example proxies would be usable for all requests)

I've updated all the modules I was able to test.

If it seems good I'll update the documentation

### CODE REVIEW

File: py3status/request.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            return self._text
+        except AttributeError:
+            if IS_PYTHON_3:
+                encoding = self._response.headers.get_content_charset('utf-8')
+            else:
+                encoding = self._response.headers.getparam('charset')
+            self._text = self._response.read().decode(encoding)
+        return self._text
+
+    def json(self):
```
Comment (ultrabug) [inline py3status/request.py]: why is text a property while json is not ?

File: py3status/py3.py [hunk 2]
```diff
@@ -787,3 +794,39 @@ def threshold_get_color(self, value, name=None):
...
+        # The aim of this function is to be a lmited lightweight replacement
+        # for the requests library but using only pythons standard libs.
+
+        # IMPORTANT NOTICE
+        # This function is excluded from private variable hiding as it is
+        # likely to need api keys etc which people may have obfuscated.
+        # Therefore it is important that no logging is done in this function
+        # that might reveal this information.
+
+        #  py3 = self
```
Comment (ultrabug) [inline py3status/py3.py]: remove me ?

**Comment by ultrabug** [comment]
A long time discussed need, thanks a lot for working on this !

File: py3status/request.py (hunk 1 above)
Comment (tobes) [inline py3status/request.py]: it is the same way that requests does things.  It also meant less changes to modules.  I'm happy to change it if you'd prefer a property

File: py3status/py3.py (hunk 2 above)
Comment (tobes) [inline py3status/py3.py]: sorry it is from my debugging

File: py3status/request.py (hunk 1 above)
Comment (ultrabug) [inline py3status/request.py]: okay makes sense

**Comment by ultrabug** [comment]
Thanks a lot !

**Pull request merged**